model/secrets_server.py

```diff
@@ -73,3 +73,15 @@ def concourse_cfg_name(self):
 
     def encrypted_concourse_cfg_name(self):
         return f'{self.encrypted_concourse_secret_name()}/{self.encrypted_concourse_attribute()}'
+
+    def secret_url_path(self, job_mapping=None, secret_cfg=None):
```

Comment:
the default for `job_mapping` seems strange to me:
- if secret_cfg is given, `None.secrets_repo()` will be accessed
- if secret_cfg is not given, `job_mapping` is not needed.

Also, a valid `secret_cfg` can effectively be anything truthy. Maybe something like
```
def secret_url_path(self, use_secret_cfg: bool = False, job_mapping = None):
  if use_secret_cfg and not job_mapping:
    raise RuntimeError('…')
…
```
That'd be a bit more precise in the signature and more expressive if it is used wrong.

The way this function is currently written I'd expect secrets_cfg to include the job_mapping and be accessed through it. Might be worth thinking about introducing an encapsulating type. I'll need to think about it some more.

Reply:
yeah, we should add typehints

Reply:
@AndreasBurger (/cc @jkrayl ): I think the idea would be to _always_ require both cfg-elements as an input